src/Fixers/ClientUpgradeFixer/ClientVar.php

```diff
@@ -0,0 +1,161 @@
+    }
+
+    public function getRpcMethod(string $rpcName): ?RpcMethod
+    {
+        // Get the Request class name
+        $newClientClass = $this->getNewClassname();
+        if (!method_exists($newClientClass, $rpcName)) {
+            // If the new method doesn't exist, there's nothing we can do
+            return null;
+        }
+        $method = new ReflectionMethod($newClientClass, $rpcName);
+        $parameters = $method->getParameters();
+        if (!isset($parameters[0]) || !$type = $parameters[0]->getType()) {
+            return null;
+        }
+        if ($type->isBuiltin()) {
+            // If the first parameter is a primitive type, assume this is a helper method
+            return null;
+        }
+
+        return new RpcMethod($this, $rpcName);
+    }
+
+    public function getLineStart(Tokens $tokens): int
+    {
+        // determine the indent
+        $indent = '';
+        $lineStart = $this->startIndex;
+        $i = 1;
+        while (
+            $this->startIndex - $i >= 0
+            && false === strpos($tokens[$this->startIndex - $i]->getContent(), "\n")
+            && $tokens[$this->startIndex - $i]->getId() !== T_OPEN_TAG
+        ) {
+            $i++;
+        }
+
+        return $this->startIndex - $i;
+    }
+
+    public static function getClientVarsFromNewKeyword(Tokens $tokens, array $clientShortNames): array
+    {
+        $clientVars = [];
+        foreach ($tokens as $index => $token) {
+            // get variables which are set directly
+            if ($token->isGivenKind(T_NEW)) {
+                $nextToken = $tokens[$tokens->getNextMeaningfulToken($index)];
+                $shortName = $nextToken->getContent();
+                if (in_array($shortName, $clientShortNames)) {
+                    if ($prevIndex = $tokens->getPrevMeaningfulToken($index)) {
+                        if ($tokens[$prevIndex]->getContent() === '=') {
+                            if ($prevIndex = $tokens->getPrevMeaningfulToken($prevIndex)) {
+                                if (
+                                    $tokens[$prevIndex]->isGivenKind(T_VARIABLE)
+                                    || (
+                                        $tokens[$prevIndex]->isGivenKind(T_STRING)
+                                        && $tokens[$prevIndex-1]->isGivenKind(T_OBJECT_OPERATOR)
+                                    )
+                                 ) {
+                                    // Handle clients set to $var
+                                    $clientClass = array_search($shortName, $clientShortNames);
+                                    $varName = $tokens[$prevIndex]->getContent();
+                                    $clientVars[$varName] = new ClientVar($varName, $clientClass);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
I think this should be changed to avoid having such nested indentation.

I would suggest something along:
```php
foreach ($tokens as $index =>  $token) {
    if (!$token->isGivenKind(T_NEW) {
        continue;
    }

    $nextToken = $tokens[$tokens->getNextMeaningfulToken($index)];
    $shortName = $nextToken->getContent();

    if (!in_array($shortName, $clientShortNames)) {
        continue;
    }

    if (!$prevIndex = $tokens->getPrevMeaningfulToken($index)) {
        continue;
    }

    ...
}

Reply:
![image](https://github.com/GoogleCloudPlatform/php-tools/assets/9062626/3d63572a-3366-4b2d-b491-af667aa3f1f3)

Reply:
This is hilarious and I'm definitely using it in the future :)